### PR TITLE
add dirtyPayload() and hide others

### PR DESCRIPTION
**Pull request by huan**
This PR is for supporting https://github.com/wechaty/wechaty-puppet-hostie/issues/43

And we have another discussion related to this PR at https://github.com/Chatie/grpc/pull/79

### CODE REVIEW

**Comment by huan** [comment]
@windmemory Please help me to review this PR and let me know what you think.

Thank you very much!

File: tests/fixtures/puppet-test/puppet-test.ts [hunk 1]
```diff
@@ -168,9 +168,16 @@ export class PuppetTest extends Puppet {
...
+  // private reset (reason: string): void {
+  //   // return super.reset(reason)
+  //   // FIXME: Huan(202008)
+  //   //  this method has overwrite the super.reset() without any functionality
+  //   //  need to change to super.reset()
+  //   //  however, we need to solve
+  //   //    `error TS1034: 'super' must be followed by an argument list or member access.`
+  //   //  first.
+  //   void reason
+  // }
```
[the remark this answers is not in the document]
Comment (huan) [reply tests/fixtures/puppet-test/puppet-test.ts:180]: Of course. Will do it later after this PR, now let's only focus on the `dirty` related code to this PR.

**Review by windmemory — APPROVED** [review]
The code looks good to me, LGTM!

**Comment by huan** [comment]
Thank you very much!

**Comment by huan** [comment]
Link to https://github.com/wechaty/wechaty-puppet-hostie/issues/43

**Pull request merged**